### ISSUE TITLE
LearningRateMonitor callback doesn't work with manual optimization 

### ISSUE THREAD

**Issue by nikvaessen** [Lightning-AI/lightning#15777]
### Bug description

When setting `self.automatic_optimization` to `False` in a `LightningModule`, a `LearningRateMonitor` callback will not log the learning rate, even though `lr_schedule.step()` is called.

### How to reproduce the bug
https://colab.research.google.com/drive/16wvrmKqGtVrOCd8Uo3K83WLeIATChFbF?usp=sharing

### Error messages and logs

See colab

### Environment

See colab

### More info

It seems `trainer._logger_connector.should_update_logs` at https://github.com/Lightning-AI/lightning/blob/8266300b29e14d719945e153dcb18743c0449966/src/pytorch_lightning/callbacks/lr_monitor.py#L151 is `False`, and therefore no logging takes place. 

cc @carmocca @nbcsm @guotuofeng

**Comment by Borda** [comment]
@carmocca mind having a look? :rabbit:
